### PR TITLE
interop: implement System.Contract.CallWithVersion

### LINKED ISSUE #2688: Implement `CallWithVersion` in interops
See https://github.com/neo-project/neo/issues/2788#issuecomment-1179524959.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2688.

### CODE REVIEW

**Comment by roman-khimov** [comment]
OK, maybe I should've written it in #2688 more specifically, but the idea is _not_ to add a new syscall, but rather handle it in the library, because it's pretty simple to do so (provide functionality, but don't introduce new calls).

**Comment by AnnaShaleva** [comment]
> I should've written it in https://github.com/nspcc-dev/neo-go/issues/2688 more specifically

It was written OK, I just didn't read the discussion carefully. Fixed, waiting for #2698 to be merge. After that I'll rebase onto master and update interop and neo-go version.

File: pkg/interop/util/util.go [hunk 1]
```diff
@@ -36,3 +38,19 @@ func Equals(a, b interface{}) bool {
 func Remove(slice interface{}, i int) {
 	neogointernal.Opcode2NoReturn("REMOVE", slice, i)
 }
+
+// CallWithVersion is a utility function that xecutes the previously deployed
+// blockchain contract with the specified version (update counter) and hash
+// (20 bytes in BE form) using the provided arguments and call flags. It fails
+// if the contract has version mismatch. It returns whatever this contract
+// returns. This function uses `System.Contract.Call` syscall.
+func CallWithVersion(scriptHash interop.Hash160, version int, method string, f contract.CallFlag, args ...interface{}) interface{} {
```
Comment (roman-khimov) [inline pkg/interop/util/util.go:47]: Move it to `pkg/interop/lib/contract`? `util` just wraps some opcodes.

Comment (AnnaShaleva) [reply pkg/interop/util/util.go]: Yes, fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**